src/main/java/bio/terra/cli/service/WorkspaceManagerServiceGcp.java

```diff
@@ -558,6 +562,70 @@ public void updateControlledGcpNotebook(
         "Error updating controlled GCP notebook in the workspace.");
   }
 
+  /**
+   * Call the Workspace Manager POST
+   * "/api/workspaces/v1/{workspaceId}/resources/controlled/gcp/dataproc-clusters/{resourceId}"
+   * endpoint to update a GCP Dataproc cluster controlled resource in the workspace.
+   *
+   * @param workspaceId the workspace where the resource exists
+   * @param resourceId the resource id
+   * @param updateParams resource properties to update
+   */
+  public void updateControlledDataprocCluster(
+      UUID workspaceId, UUID resourceId, UpdateControlledGcpDataprocClusterParams updateParams) {
+
+    ControlledGcpResourceApi gcpResourceApi = new ControlledGcpResourceApi(apiClient);
+    String updateErrMsg = "Error updating controlled GCP Dataproc cluster in the workspace";
+
+    // Update WSM metadata fields and primary, secondary worker counts, and graceful decommission
+    // timeout.
+    UpdateControlledGcpDataprocClusterRequestBody updateRequest =
+        new UpdateControlledGcpDataprocClusterRequestBody()
+            .name(updateParams.resourceFields.name)
+            .description(updateParams.resourceFields.description);
+
+    updateRequest.updateParameters(
+        new ControlledDataprocClusterUpdateParameters()
+            .numPrimaryWorkers(updateParams.numWorkers)
+            .numSecondaryWorkers(updateParams.numSecondaryWorkers)
+            .gracefulDecommissionTimeout(updateParams.gracefulDecommissionTimeout));
+    callWithRetries(
+        () -> gcpResourceApi.updateDataprocCluster(updateRequest, workspaceId, resourceId),
+        updateErrMsg);
+
+    // Update autoscaling policy independently. Dataproc api does not allow autoscaling policy and
+    // other attributes in tandem.
+    if (updateParams.autoscalingPolicyUri != null) {
+      UpdateControlledGcpDataprocClusterRequestBody autoscalingPolicyUpdateRequest =
+          new UpdateControlledGcpDataprocClusterRequestBody()
+              .updateParameters(
+                  new ControlledDataprocClusterUpdateParameters()
+                      .autoscalingPolicy(updateParams.autoscalingPolicyUri));
+      callWithRetries(
+          () ->
+              gcpResourceApi.updateDataprocCluster(
+                  autoscalingPolicyUpdateRequest, workspaceId, resourceId),
+          updateErrMsg);
+    }
+
+    // Update scheduled deletion independently. Dataproc api does not allow autoscaling policy and
```

Comment:
Dataproc api does not allow updating idleDeleteTtl and other attributes together